### PR TITLE
feat(types): Export most of the types

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

Expose most of the types in the `xrpl` package.

BREAKING CHANGE: Renames of deep importable types:
- `methods/accountLines`: `Trustline` -> `AccountLinesTrustline`
- `methods/bookOffers`: `TakerAmount` -> `BookOfferCurrency`
- `methods/ledgerData`: `BinaryLedgerEntry` -> `LedgerDataBinaryLedgerEntry`

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation Updates

### CODE REVIEW

**Review by justinr1234 — APPROVED** [review]

File: packages/xrpl/src/models/ledger/NFTokenPage.ts [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
+    NFTokenTaxon: number
+    URI?: string
+  }
 }
 
 export interface NFTokenPage extends BaseLedgerEntry {
   LedgerEntryType: 'NFTokenPage'
   NextPageMin?: string
-  NFTokens: NFTokenWrapper[]
+  NFTokens: NFToken[]
```
Comment (JST5000) [inline packages/xrpl/src/models/ledger/NFTokenPage.ts:16]: Why was this changed from `NFTokenWrapper`?

Comment (ckniffen) [reply packages/xrpl/src/models/ledger/NFTokenPage.ts:16]: To keep it inline with the other nested STObjects in STArrays.  After looking at more definitions i decided to change it.  NFTokenWrapper was never in a released version of xrpl.js so figured now would be a good time to change it.

**Review by khancode — APPROVED** [review]
